pkg/innerring/processors/netmap/cleanup_table.go

```diff
@@ -36,33 +44,50 @@ func (c *cleanupTable) update(snapshot *netmap.Netmap, now uint64) {
 	defer c.Unlock()
 
 	// replacing map is less memory efficient but faster
-	newMap := make(map[string]epochStamp, len(snapshot.Nodes))
+	newMap := make(map[string]epochStampWithNodeInfo, len(snapshot.Nodes))
 
 	for i := range snapshot.Nodes {
+		binNodeInfo, err := snapshot.Nodes[i].Marshal()
+		if err != nil {
+			panic(fmt.Errorf("could not marshal node info: %w", err)) // seems better than ignore
+		}
+
 		keyString := hex.EncodeToString(snapshot.Nodes[i].PublicKey())
-		if access, ok := c.lastAccess[keyString]; ok {
+
+		access, ok := c.lastAccess[keyString]
+		if ok {
 			access.removeFlag = false // reset remove Flag on each Update
-			newMap[keyString] = access
 		} else {
-			newMap[keyString] = epochStamp{epoch: now}
+			access.epoch = now
 		}
+
+		access.binNodeInfo = binNodeInfo
+
+		newMap[keyString] = access
 	}
 
 	c.lastAccess = newMap
 }
 
-func (c *cleanupTable) touch(keyString string, now uint64) bool {
+// updates last access time of the netmap node by string public key.
+//
+// Returns true if node hasn't been accessed yet, or:
+//  * remove flag is unset, and
+//  * binary node info is changed.
+func (c *cleanupTable) touch(keyString string, now uint64, binNodeInfo []byte) bool {
 	c.Lock()
 	defer c.Unlock()
 
 	access, ok := c.lastAccess[keyString]
-	result := !access.removeFlag && ok
+	result := !ok || !access.removeFlag && !bytes.Equal(access.binNodeInfo, binNodeInfo)
```

Comment:
```suggestion
	result := !ok || access.removeFlag || !bytes.Equal(access.binNodeInfo, binNodeInfo)
```

If `removeFlag` is set, then `touch()` always has to return `true` to resign transaction.
If `binNodeInfo` has been changed, then `touch()` always has to return true to resign transaction.

Reply:
Fixed